classes/Language.php

```diff
@@ -1253,6 +1241,9 @@ public static function downloadXLFLanguagePack($locale, &$errors = [], $type = s
             return false;
         }
 
+        $language_pack_loader = RemoteLanguagePackLoader::build();
```

Comment:
I'm not a fan of introducing a static function in a service class, this encourages using this method in favor of the Symfony DI which is not the goal

When legacy code wants to use a Symfony service outside of the framework then it's its responsibility to handle the fallback so you should manually create the object here. Besides you removed the `$version` parameter in favor of a hard-coded one in the static build function, but since the service accepts a version parameter I see no reason why we should remove this parameter which induces a big breaking change

```suggestion
        $languagePackLoader = new RemoteLanguagePackLoader(new Version($version));
```

Reply:
I think you missed something.

The `$version` parameter I removed in `downloadAndInstallLanguagePack` wasn't used at all.
In this function (`downloadXLFLanguagePack`), the parameter I removed is `$type` which was used for email/language distinction.

In this function, version used environment variable (`_PS_VERSION_`)
```php
$url = (self::PACK_TYPE_EMAILS === $type) ? self::EMAILS_LANGUAGE_PACK_URL : self::SF_LANGUAGE_PACK_URL;
$url = str_replace(
    [
        '%version%',
        '%locale%',
    ],
    [
        _PS_VERSION_,
        $locale,
    ],
    $url
);
```

So if you really don't like the `build` function I can replace it everywhere by a 
```
$languagePackLoader = new RemoteLanguagePackLoader(new Version(_PS_VERSION_));
```

So let me know If you prefer hard-coded version for every use of the `RemoteLanguagePackLoader` and I remove the static `build()` function or if it's Ok to keep `build()` with those elements.

In my head, using a function instead of environment variable helps for maintainability due to encapsulation principle. But in this case I agree it doesn't matter that much. `_PS_VERSION_` shouldn't change, it's internal set parameter and it would be easy to find and change if necessary (even with all the code dependent on it). The only case I see it would be a problem is if you extract a code to make a library or something independent.